<launch>
	<arg name="image_topic_in"  />
	<arg name="image_topic_out" 	default="$(arg image_topic_in)/pipelined" />
	<arg name="nodelet_manager" 	default="nodelet_manager" />
	
	<include file="$(find image_enhancement)/launch/blur.launch" >
  		<arg name="name" 					value="image_enhancement_blur" />
		<arg name="image_topic_in"			value="$(arg image_topic_in) />
		<arg name="image_topic_out" 		value="$(arg image_topic_in)_blur" />
		<arg name="nodelet_manager" 		value="$(arg nodelet_manager)" />
  	</include>
	
	<include file="$(find image_enhancement)/launch/gamma.launch" >
  		<arg name="name" 					value="image_enhancement_gamma" />
		<arg name="image_topic_in"			value="$(arg image_topic_in)_blur" />
		<arg name="image_topic_out" 		value="$(arg image_topic_in)_blur_gamma" />
		<arg name="nodelet_manager" 		value="$(arg nodelet_manager)" />
		<arg name="gamma"					value="2.2" />
  	</include>
  	
  	<include file="$(find image_enhancement)/launch/dilate.launch" >
  		<arg name="name" 					value="image_enhancement_dilate" />
		<arg name="image_topic_in"			value="$(arg image_topic_in)_blur_gamma" />
		<arg name="image_topic_out" 		value="$(arg image_topic_in)_blur_gamma_dilate" />
		<arg name="nodelet_manager" 		value="$(arg nodelet_manager)" />
		<arg name="dilation_size"			value="1" />
  	</include>
  	
  	<include file="$(find image_enhancement)/launch/erode.launch" >
  		<arg name="name" 					value="image_enhancement_erode" />
		<arg name="image_topic_in"			value="$(arg image_topic_in)_blur_gamma_dilate" />
		<arg name="image_topic_out" 		value="$(arg image_topic_out)" />
		<arg name="nodelet_manager" 		value="$(arg nodelet_manager)" />
		<arg name="erosion_size"			value="1" />
  	</include>
	  
</launch>